agent-system/src/SystemAgent.cc

```diff
@@ -332,7 +359,8 @@ SystemAgent::Read (const YCPPath& path, const YCPValue& arg, const YCPValue&)
 	 * @example Read (.target.tmpdir) -> "/some/temp/dir"
 	 */
 
-	return YCPString (tempdir);
+	// skip the chroot prefix, return the location relative to the current root
+	return YCPString (tempdir().substr(strlen(root())));
```

Comment:
I think this is wrong :)

It is fine when `tempdir` is `/mnt/tmp` and `root` is `/mnt`
It is fine when `tempdir` is `/tmp` and `root` is  empty
But what if `root` is `/`?

AFAIK SCRAgent::root is not clear about "" vs "/"

Yet your test shows it works... ... ah, because `tempdir()` will return two leading slashes. I think it deserves a comment in the code.

Reply:
for safety I would also change code to something like (pseudocode) `YCPString ((root() == "/" || root() == "") ? tempdir() : tempdir().substr(strlen(root())))`